### PR TITLE
M2P-85 Create an offline Magento invoice and move order status to processing after placing an order using a saved card in the back-office

### PR DESCRIPTION
**Pull request by johnmccombs1**
# Description
Currently, after the admin creates an order using a saved card in the back-office, the order is created and stuck in the pending status as the Bolt recharged transaction doesn’t have webhooks.

This PR creates an offline Magento invoice for the recharged order and move its status to processing.

Fixes: https://boltpay.atlassian.net/browse/M2P-85

#changelog Create an offline Magento invoice and move order status to processing after placing an order using a saved card in the back-office

# Type of change

- [ ] Bug fix (change which fixes an issue)
- [x] New feature (change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] This change requires a documentation update

# How Has This Been Tested?
Please validate that you have tested your change in at least one of the following areas:

- [x] Successfully tested locally (or docker image)
- [ ] Successfully tested on a staging or sandbox server
- [ ] Successfully tested on a merchant's staging server

# Checklist:

- [x] My code follows the style guidelines of this project.
- [ ] I have performed a self-review of my own code.
- [ ] I have commented my code, particularly in hard-to-understand areas.
- [ ] New and existing unit tests pass locally with my changes.
- [ ] I have created or modified unit tests to sufficiently cover my changes.
- [ ] I have added my Jira ticket link and provided a changelog message.

### CODE REVIEW

**Comment by johnmccombs1** [comment]
@danac-gs  because the recharged transaction is a direct payment. Bolt doesn’t authorize and capture it. It just goes to complete status right away after its creation. It’s just like an auto-capture transaction without the hooks being sent to Magento.

File: Plugin/OrderPlugin.php [hunk 1]
```diff
@@ -48,6 +63,11 @@ public function beforeSetState(Order $subject, $state)
         if (!$subject->getPayment() || $subject->getPayment()->getMethod() != \Bolt\Boltpay\Model\Payment::METHOD_CODE) {
             return [$state];
         }
+
+        if ($subject->getIsRechargedOrder()){
```
Comment (vitaliyreznikov) [inline Plugin/OrderPlugin.php:67]: Could you add a comment while we need this?

Comment (johnmccombs1) [reply Plugin/OrderPlugin.php:67]: @vitaliyreznikov I've updated

File: Plugin/OrderPlugin.php [hunk 2]
```diff
@@ -69,6 +90,12 @@ public function beforeSetStatus(Order $subject, $status)
         if (!$subject->getPayment() || $subject->getPayment()->getMethod() != \Bolt\Boltpay\Model\Payment::METHOD_CODE) {
             return [$status];
         }
+
+        if ($subject->getIsRechargedOrder()) {
+            // Check and set the recharged order state to processing
```
Comment (vitaliyreznikov) [inline Plugin/OrderPlugin.php:95]: Could you add a comment while we need set status to processing in this case? What happens if we don't do this?

**Pull request merged**